src/smart-components/role/edit-resource-definitions-modal.js

```diff
@@ -131,19 +140,21 @@ const EditResourceDefinitionsModal = ({ cancelRoute }) => {
     }
   };
 
-  const handleSubmit = (data, options) => {
+  const handleSubmit = (data) => {
     dispatchLocally({ type: 'update', payload: { changedResources: data['dual-list-select'] } });
     const newAccess = {
       permission: permissionId,
-      resourceDefinitions: data['dual-list-select'].map((value) => ({
-        attributeFilter: {
-          key: `cost-management.${options.find((option) => option.value === value).path}`,
-          operation: 'equal',
-          value,
+      resourceDefinitions: [
+        {
+          attributeFilter: {
+            key: `cost-management.${permissionId.split(':')?.[1]}`,
+            operation: data['dual-list-select'].length === 1 ? 'equal' : 'in',
+            value: data['dual-list-select'].length === 1 ? data['dual-list-select'][0] : data['dual-list-select'],
+          },
         },
-      })),
+      ],
     };
-    dispatch(updateRole(roleId, { ...role, access: role.access.map((item) => (item.permission === permissionId ? newAccess : item)) }), true).then(
+    dispatch(updateRole(roleId, { ...role, access: [...role.access.filter((item) => item.permission !== permissionId), newAccess] }), true).then(
```

Comment:
Are you sure that this is working when you are sending only changed resources? That it doesn't delete the unchaged ones.

Reply:
I am pretty sure That you have to include the unchaged resources, otherwise they get deleted

Reply:
I think I am passing them, or not?
`...role.access.filter((item) => item.permission !== permissionId)`

Reply:
Yep, you are right. In that case it it fine.